web/src/components/page.jsx

```diff
@@ -1,20 +1,46 @@
 import React from 'react';
 
 import Header from '../containers/header';
+import HeaderBtn from './header-btn';
+
+let DEFAULT_STATE = {
+  headerIsOpen: true,
+}
 
 export default class Page extends React.Component {
   constructor(props) {
     super(props);
+    this.state = DEFAULT_STATE;
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState((prevState) => {
+      return { headerIsOpen: !prevState.headerIsOpen }
+    });
   }
 
   render() {
-    return [
-      <Header key="header" />,
-      <div id="page" key="page">
-        <main key="main">
-          {this.props.children}
-        </main>
-      </div>
-    ];
+    if (this.state.headerIsOpen) {
+      return [
+        <Header key="header" />,
+        <div id="page" key="page">
+          <HeaderBtn arrow="header-btn-arrow" isOpen={this.state.headerIsOpen} handleClick={this.handleClick} />
+          <main key="main">
+            {this.props.children}
+          </main>
+        </div>
+      ];
+
+    } else {
+      return [
+        <div id="page" key="page">
+          <HeaderBtn arrow="header-btn-arrow header-btn-arrow-right" handleClick={this.handleClick} />
```

Comment:
as the presence of `header-btn-arrow-right` depends on `headerIsOpen`, I guess we could directly do this in `header-button.jsx`, similar to the already existing check we have there for the class name. Then we wouldn't need to pass `arrow` at all, and the `page.jsx` does not care at all if the menu is open or not in terms of styling. I think it would be good to encapsulate this, and then go further and handle the state in `header-button.jsx` as well. With these changes, `page.jsx` always has a `HeaderBtn`, but doesn't care whether it's open or not.

Reply:
Hi @MichaelKohler ! I managed to add the conditional for the arrow into the `header-btn.jsx`. Can you please explain a little more about the "encapsulating" part? The `Header-Btn` component should hide the header?

Reply:
Depending on how you fixed the first issue, this might not be necessary anymore. Can you link me to your latest changes on GitHub so I can have another look?

Reply:
(or create a PR, would work for me as well)